### PR TITLE
[FIX]  Compatibility with VTK 9

### PR DESCRIPTION
**Pull request by skoudoro**
This PR update some FURY functions due to some changes in VTK 9. The main change is better management of vtkCellArray.

Wheels available on Pypi only for Windows and macOS. Concerning Linux, you have to download the wheel on VTK website. (https://discourse.vtk.org/t/vtk-9-wheels-on-pypi/3340/4)

VTK Releases Announcement :  https://blog.kitware.com/vtk-9-0-0-available-for-download/
VTK API changes from 8.2 to 9.0: https://vtk.org/Wiki/VTK/API_Changes_8_2_0_to_9_0_0
VTK 9 build process : https://vtk.org/doc/nightly/html/md__home_kitware_dashboards_buildbot_vtk_nightly-master-ike-linux-shared-release_doc_nightly_osm749dd663df9981384f2598c108aac3b0.html

### CODE REVIEW

File: fury/actor.py [hunk 1]
```diff
@@ -1204,20 +1212,21 @@ def _tensor_slicer_mapper(evals, evecs, affine=None, mask=None, sphere=None,
...
+    if vtk.vtkVersion.GetVTKMajorVersion() >= 9:
+        cells = numpy_to_vtk_cells(all_faces, is_coords=False)
+    else:
+        all_faces = np.hstack((3 * np.ones((len(all_faces), 1)),
+                              all_faces))
+        ncells = len(all_faces)
+        all_faces = np.ascontiguousarray(all_faces.ravel(), dtype='i8')
+        all_faces_vtk = numpy_support.numpy_to_vtkIdTypeArray(all_faces,
+                                                              deep=True)
+        cells.SetCells(ncells, all_faces_vtk)
```
Comment (arokem) [inline fury/actor.py:1229]: Might be better to make a function out of this, so you don't have to repeat this code. 

The easiest for maintenance in the future would probably be to make one function that encompasses the different aspects handled in this PR, but that might end up a bit of a multi-headed monster.

Comment (skoudoro) [reply fury/actor.py:1229]: I agree. In general, `tensor_slicer` and `odf_slicer`function need some refactoring. A lot of code is duplicated.

To answer your proposition, This is why I created `numpy_to_vtk_cells` function but it seems it needs a bit more thinking to manage more cases.  I will update that.

Thank you for the quick review!

**Comment by skoudoro** [comment]
I have installed 3 versions of VTK on my mac and I cannot reproduce the error on Travis macOS. So I propose to temporarily skip this test on mac until we find a solution.

**Comment by skoudoro** [comment]
Ok, This PR is ready to go. Can you have a look and merge it @Garyfallidis?  @MarcCote? @dmreagan?  @guaje? @arokem?

File: fury/tests/test_actors.py [hunk 2]
```diff
@@ -310,26 +311,32 @@ def test_contour_from_roi():
         # window.show(r2)
 
 
-def test_contour_from_label():
+@pytest.mark.skipif(skip_osx, reason="This test does not work on macOS + "
+                                     "Travis. It works on a local machine"
+                                     " with 3 different version of VTK. There"
+                                     " is 2 problem to check: Travis macOS vs"
```
Comment (MarcCote) [inline fury/tests/test_actors.py:317]: there is -> there are 
problem -> problems

Comment (skoudoro) [reply fury/tests/test_actors.py:317]: Thanks for the feedback!

**Comment by Garyfallidis** [comment]
@arokem and @MarcCote in your current review. Have you installed VTK 9 in your computer and checked the result?

**Comment by arokem** [comment]
No. I have not had a chance to give this a try. My code was just about code duplication/design.

**Comment by Garyfallidis** [comment]
@skoudoro all we need to do is ``pip install --upgrade vtk``. Correct?

**Comment by skoudoro** [comment]
Yes @Garyfallidis, but only on Windows and macOS.  Concerning Linux, you have to download the wheel on VTK website. We need to ask them to add the wheel for Linux on PyPI too

**Comment by Garyfallidis** [comment]
Next time add this information in the description of the PR. That will help reviewers a lot.

**Comment by Garyfallidis** [comment]
Can the two reviewers upgrade vtk and try the code? I will do the same here. In the meantime @skoudoro can you ask VTK to add the Linux wheels?

**Comment by skoudoro** [comment]
they are already working on it as you can see [here](https://discourse.vtk.org/t/vtk-9-wheels-on-pypi/3340/4)

**Comment by Garyfallidis** [comment]
Okay so it should appear online relatively soon. 👍

**Comment by MarcCote** [comment]
I just installed VTK9 and the `test_grid` test hangs on my machine. The window does not close itself. Maybe there's an issue with the timer.

When running the `test_ui.py::test_timer` I got the following:
```
fury/tests/test_ui.py::test_timer X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  25 (X_SendEvent)
  Resource id in failed request:  0x200002
  Serial number of failed request:  156
  Current serial number in output stream:  163
```
also, still in `test_ui.py::test_timer`, it seems the count only increases when I moved the mouse over the window.

**Comment by skoudoro** [comment]
Thank you for the feedback. Was it on Linux @MarcCote? Offscreen or not?

This PR is really important. I will need to release FURY next week and unblock DIPY PR.
Did you try it @Garyfallidis @arokem ?

**Comment by MarcCote** [comment]
Yes, on LinuxMint 19.3. I downloaded the VTK wheel and installed it from VTK Github release page. I tried both offscreen and onscreen.

**Comment by skoudoro** [comment]
ok, Thank you

**Comment by skoudoro** [comment]
VTK Releases Announcement :  https://blog.kitware.com/vtk-9-0-0-available-for-download/
VTK API changes from 8.2 to 9.0: https://vtk.org/Wiki/VTK/API_Changes_8_2_0_to_9_0_0
VTK 9 build process : https://vtk.org/doc/nightly/html/md__home_kitware_dashboards_buildbot_vtk_nightly-master-ike-linux-shared-release_doc_nightly_osm749dd663df9981384f2598c108aac3b0.html

**Comment by MarcCote** [comment]
@Garyfallidis @arokem I'm curious to know if the timer issue with VTK9 on Linux is only happening to me?

**Comment by MarcCote** [comment]
FYI: I've opened an issue about my issue: https://gitlab.kitware.com/vtk/vtk/-/issues/17918

**Comment by skoudoro** [comment]
Thank you for this @MarcCote  ! Looking forward for their feedback

**Comment by MarcCote** [comment]
Small update. One of the VTK developers found the issue and will issue a fix.

**Comment by skoudoro** [comment]
So it means we should recommend the users to avoid vtk 9 until they have a release patch. What do you think @MarcCote? I will update our requirements.txt

**Comment by MarcCote** [comment]
Yes, that's might be better. I don't know how long making a patch release takes for VTK: https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6972

**Comment by MarcCote** [comment]
They are planning to have a patch release next week. https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6972#note_773691

**Comment by skoudoro** [comment]
Great ! We need to check if there is no other critical issue before this patch release. We should find a way to test it. I think we have no choice to get the source and compile it.

But that's an excellent news!

**Comment by MarcCote** [comment]
Also, `pytest -svv fury/tests/test_window.py::test_order_transparent` is failing on VTK9.
Here's the screenshot corresponding to the following line in the test
```python
    arr = window.snapshot(scene, fname='green_front.png',
                          offscreen=True, order_transparent=False)
```
green_front

compared to what we had for VTK <9
green_front

**Comment by Garyfallidis** [comment]
Thank you @MarcCote and @skoudoro for your useful contribution here. @MarcCote be happy to move your last comment to a new issue. I merging this here to facilitate further development and fixes for vtk 9.

**Comment by MarcCote** [comment]
@Garyfallidis Is VTK9 working on your Linux machine? At the very least this should have been merged as well: https://github.com/skoudoro/fury/pull/4

**Comment by skoudoro** [comment]
Sorry @MarcCote, I was not aware of this PR. I didn't get any notification from my own fork. I just see it. Do you want to rebase this and create a PR or should I do it. 

PS: I will add notification on my fork

**Pull request merged**